mrmustard/physics/wires.py

```diff
@@ -1,31 +1,169 @@
 
-from mrmustard import widgets
+    def __str__(self) -> str:
+        return self.name
 
-__all__ = ["Wires"]
+    def __repr__(self) -> str:
+        return self.name
+
+
+class ReprEnum(LegibleEnum):
+    """Enumeration of possible representations for quantum states and operations."""
+
+    UNSPECIFIED = auto()
+    BARGMANN = auto()
+    FOCK = auto()
+    QUADRATURE = auto()
+    PHASESPACE = auto()
+    CHARACTERISTIC = auto()
+
+
+class WiresType(LegibleEnum):
+    """Enumeration of possible wire types in quantum circuits."""
+
+    DM_LIKE = auto()  # only output ket and bra on same modes
+    KET_LIKE = auto()  # only output ket
+    UNITARY_LIKE = auto()  # such that can map ket to ket
+    CHANNEL_LIKE = auto()  # such that can map dm to dm
+    PROJ_MEAS_LIKE = auto()  # only input ket
+    POVM_LIKE = auto()  # only input ket and input bra on same modes
+    CLASSICAL_LIKE = auto()  # only classical wires
+
+
+@dataclass(slots=True)
+class QuantumWire:
+    """
+    Represents a quantum wire in a circuit.
+
+    Args:
+        mode: The mode number this wire represents
+        is_out: Whether this is an output wire
+        is_ket: Whether this wire is on the ket side
+        index: The index of this wire in the circuit
+        repr: The representation of this wire
+        id: Unique identifier for this wire
+    """
+
+    mode: int
+    is_out: bool
+    is_ket: bool
+    index: int
+    repr: ReprEnum = ReprEnum.BARGMANN
+    repr_params: Any = None
+    id: int = field(default_factory=lambda: randint(0, 2**32 - 1), compare=False)
+
+    def __hash__(self) -> int:
+        return hash((self.mode, self.is_out, self.is_ket, self.repr))
+
+    def __repr__(self) -> str:
+        return f"QuantumWire(mode={self.mode}, {'out' if self.is_out else 'in'}, {'ket' if self.is_ket else 'bra'}, repr={self.repr}, index={self.index})"
+
+    def __eq__(self, other: QuantumWire) -> bool:
+        return (
+            self.mode == other.mode
+            and self.is_out == other.is_out
+            and self.is_ket == other.is_ket
+            and self.repr == other.repr
+        )
+
+    def copy(self, new_id: bool = False) -> QuantumWire:
+        """Create a copy of the quantum wire.
+
+        Args:
+            new_id (bool): If True, generates a new ID for the copy. Defaults to False.
+
+        Returns:
+            QuantumWire: A copy of the quantum wire
+        """
+        return QuantumWire(
+            mode=self.mode,
+            is_out=self.is_out,
+            is_ket=self.is_ket,
+            index=self.index,
+            repr=self.repr,
+            repr_params=self.repr_params,
+            id=self.id if not new_id else randint(0, 2**32 - 1),
+        )
+
+    def _order(self) -> int:
+        """
+        Artificial ordering for sorting quantum wires.
+        Order achieved is by bra/ket, then out/in, then mode.
+        """
+        return self.mode + 10_000 * (1 - 2 * self.is_out) - 100_000 * (1 - 2 * self.is_ket)
```

Comment:
This seems to be used as the sorting key.  Perhaps using lex ordering of tuples would be more readable?

Reply:
What do you mean? Can you suggest a change?